CarService/CarService.Users.Api/Controllers/UsersController.cs

```diff
@@ -55,15 +56,29 @@ public async Task<IActionResult> CreateUser([FromBody]CreateUserRequest model)
         }
 
         [HttpGet]
-        public async Task<IActionResult> GetUsers([FromQuery] Guid carId)
+        public async Task<IActionResult> GetUsers([FromQuery] Guid carId, [FromQuery] string username = null)
         {
             var dtoList = await _mediator.Send(new GetAllUsersQuery());
             IEnumerable<UserResponse> response;
             
-            if (carId == Guid.Empty)
+            if (carId == Guid.Empty && string.IsNullOrWhiteSpace(username))
             {
                 response = dtoList.Select(dto => _mapper.Map<UserResponse>(dto));
             }
+            else if (!string.IsNullOrWhiteSpace(username))
+            {
```

Comment:
This is bad practice. In case you have a lot of users, you'll first retrieve them all and then filter. It should be done on the DB side. Probably worth having separate query for this